### ISSUE TITLE
Bug: [rule name here] <short description of the issue>

### ISSUE THREAD

**Issue by gerardolima** [typescript-eslint/typescript-eslint#5650]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.3&sourceType=module&code=C4TwDgpgBACg9sAhsOUC8UDOwBOBLAOwHMAoEgegCpKoSoaAVACz0ylakQPYIDM4cAW2R4AbtABSAZQAicAMZR5cQYIgFgAGigB3PMCacoOCLwgmC86CigBvAAIAbQgGtYCZHAC+2xLXpQoJCBTMjsbIjywACuiI6OIFAECFDRmBAAJlAARokM4BBS8vhgwEpwGRAAdHSU5CQQAB5gAmXKBNhQ-KgYABQAlABcWLiEROgAfFAA5NmIONMkQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaWXWZRAE0IDcBDaVDSfAMxcgBpxtIABAFwCeABxQBjaAEth-QiniTc-APQFipclToN0YANqRE0aPmiQAutwC+IK0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type Potato = string

/** 
 * This is an informative JSDoc comment, with a reference to {@link Potato}, a 
 * type that is actually not used by TypeScript code.
 */
export const foo = (): string => 'bar'
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

The type `Potato` is used in JSDoc. I wouldn't expect eslint rules to consider it "unused".

### Actual Result

'Potato' is defined but never used. 1:6 - 1:12

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.37.0` |
| `@typescript-eslint/parser` | `5.37.0` |
| `TypeScript` | `4.8.3` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by bradzacher** [comment]
First off - when filing an issue please make sure you fill out the title as well as the issue form.

--

We don't support JSDoc comments.
This is on purpose - JSDoc is not actual code and thus it should not mark variables as used.

If you want the variable to be used - use it in your code. Eg
```ts
export const foo = (): Potato => 'bar'
```

**Comment by Josh-Cena** [comment]
@gerardolima FYI: https://github.com/gajus/eslint-plugin-jsdoc/issues/858
